bot/exts/moderation/incidents.py

```diff
@@ -146,6 +208,35 @@ async def add_signals(incident: discord.Message) -> None:
             await incident.add_reaction(signal_emoji.value)
 
 
+async def extract_message_links(message: discord.Message, bot: Bot) -> t.Optional[list]:
+    """
+    Checks if there's any message links in the text content.
+
+    Then passes the the message_link into `make_message_link_embed` to format a
+    embed for it containing information about the link.
+
+    As discord only allows a max of 10 embeds in a single webhook, just send the
+    first 10 embeds and don't care about the rest.
+
+    If no links are found for the message, it logs a trace statement.
+    """
```

Comment:
Why is this not an instance method on the cog if it uses the cog's state?

Consider making it a method and accessing `self.bot` rather than passing it as a param.

The return type annotation can be extended to be `List[Embed]`.

Reply:
Resolved in ffc39fb.